core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -1,37 +1,164 @@
+                .get();
+
+        CountDownLatch ryukScheduledLatch = new CountDownLatch(1);
+
+        synchronized (DEATH_NOTE) {
+            DEATH_NOTE.add(
+                    DockerClientFactory.DEFAULT_LABELS.entrySet().stream()
+                            .<Map.Entry<String, String>>map(it -> new SimpleEntry<>("label", it.getKey() + "=" + it.getValue()))
+                            .collect(Collectors.toList())
+            );
+        }
+
+        Thread kiraThread = new Thread(
+                () -> {
+                    while (true) {
+                        int index = 0;
+                        try(Socket clientSocket = new Socket(hostIpAddress, ryukPort)) {
+                            OutputStream out = clientSocket.getOutputStream();
+                            BufferedReader in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));
+
+                            synchronized (DEATH_NOTE) {
+                                while (true) {
+                                    if (DEATH_NOTE.size() <= index) {
+                                        try {
+                                            DEATH_NOTE.wait(1_000);
+                                            continue;
+                                        } catch (InterruptedException e) {
+                                            throw new RuntimeException(e);
+                                        }
+                                    }
+                                    List<Map.Entry<String, String>> filters = DEATH_NOTE.get(index);
+
+                                    String query = URLEncodedUtils.format(
+                                            filters.stream()
+                                                    .map(it -> new BasicNameValuePair(it.getKey(), it.getValue()))
+                                                    .collect(Collectors.toList()),
+                                            (String) null
+                                    );
 
-        // If the JVM stops without containers being stopped, try and stop the container.
-        Runtime.getRuntime().addShutdownHook(new Thread(this::performCleanup));
+                                    log.debug("Sending '{}' to Ryuk", query);
+                                    out.write(query.getBytes());
+                                    out.write('\n');
+                                    out.flush();
+
+                                    while (!"ACK".equalsIgnoreCase(in.readLine())) {
+                                    }
+
+                                    ryukScheduledLatch.countDown();
+                                    index++;
+                                }
+                            }
+                        } catch (IOException e) {
+                            log.warn("Can not connect to Ryuk at {}:{}", hostIpAddress, ryukPort, e);
+                        }
+                    }
+                },
+                "tc-ryuk"
```

Comment:
Same as above, please could you expand `tc` to `testcontainers` for googlability?